packages/components/src/entities/SamplesDeriveButton.tsx

```diff
@@ -21,8 +22,49 @@ export interface SamplesDeriveButtonProps
 
 export const SamplesDeriveButton: FC<SamplesDeriveButtonProps> = memo(props => {
     const { model, asSubMenu, ...createSampleMenuProps } = props;
+    const [selectionsAreSet, setSelectionsAreSet] = useState<boolean>(false);
+    const [selectionData, setSelectionData] = useState<Map<any, any>>();
+    const selectedCount = useMemo(() => model?.selections?.size ?? -1, [model?.selections]);
+    const useSelectionData = useMemo(() => model?.filterArray.length > 0 && selectedCount <= MAX_EDITABLE_GRID_ROWS , [model?.filterArray]);
```

Comment:
Nice! It makes much more sense to only get the selected data if we actually intend on using it.